### ISSUE TITLE
FlexibleSolver Potentially Dereferences Null-pointer

### ISSUE THREAD

**Issue by bska** [OPM/opm-simulators#1941]
I just recently got exposure to the new [`FlexibleSolver`](https://github.com/OPM/opm-simulators/blob/c1574a23d0071823a9245e9ded5fbc4ca9da0f3b/opm/simulators/linalg/FlexibleSolver.hpp) facility.  While running [`test_flexiblesolver`](https://github.com/OPM/opm-simulators/blob/c1574a23d0071823a9245e9ded5fbc4ca9da0f3b/tests/test_flexiblesolver.cpp) in a checked iterator debug build I got a null-pointer dereference error in  [`PressureTransferPolicy::createCoarseLevelSystem()`](https://github.com/OPM/opm-simulators/blob/c1574a23d0071823a9245e9ded5fbc4ca9da0f3b/opm/simulators/linalg/PressureTransferPolicy.hpp#L71), specifically the sub-expression
```C++
*coarseLevelCommunication_
```
dereferences `null` in this case.

It is entirely possible that this is just a local issue, or alternatively, a problem in the unit test.  I'll nevertheless record the potential issue here.

**Comment by atgeirr** [comment]
You are right there is a bug: the serial (non-comm) constructor for OwningTwoLevelPreconditioner does this in its initializer list:
```
...
        , comm_(nullptr)
...
        , levelTransferPolicy_(*comm_, weights_, prm.get<int>("pressure_var_index"))
```

**Comment by atgeirr** [comment]
Hopefully fixed by #1962, closing.
